build/publish.proj

```diff
@@ -37,6 +36,10 @@
     <Error Condition="'$(BUILD_REPOSITORY_URI)' == ''" Text="The BUILD_REPOSITORY_URI property is required." />
     <Error Condition="'$(MaestroAccessToken)' == ''" Text="The MaestroAccessToken property is required." />
     <Error Condition="'$(MaestroApiEndpoint)' == ''" Text="The MaestroApiEndpoint property is required." />
+    <Error Condition="'$(SYSTEM_TEAMFOUNDATIONCOLLECTIONURI)' == ''" Text="The SYSTEM_TEAMFOUNDATIONCOLLECTIONURI property is required." />
+    <Error Condition="'$(SYSTEM_TEAMPROJECT)' == ''" Text="The SYSTEM_TEAMPROJECT property is required." />
+    <Error Condition="'$(BUILD_BUILDID)' == ''" Text="The BUILD_BUILDID property is required." />
+    <Error Condition="'$(SYSTEM_DEFINITIONID)' == ''" Text="The SYSTEM_DEFINITIONID property is required." />
```

Comment:
This part is added for better debugging by yourself, right?

Reply:
No, it's error detection/handling. [In the previous PR JohnTortugo suggested copying what the Arcade SDK does](https://github.com/NuGet/NuGet.Client/pull/3302#discussion_r394582815). I was just too lazy to explicitly set these properties on the command line in the YAML, and letting them come in implicitly from environment variables.

Reply:
Thanks for clarification!